docs/source/common_issues.rst

```diff
@@ -532,3 +532,28 @@ put the linter comment *after* the type comment:
 .. code-block:: python
 
     a = some_complex_thing()  # type: ignore  # noqa
+
+
+Dealing with conflicting names
+------------------------------
+
+Suppose you have a class with a method whose name is the same as an
+imported (or built-in) type, and you want to use the type in another
+method signature.  E.g.
+
+.. code-block:: python
+
+   class C:
```

Comment:
I would give these more descriptive name. For example, `C` -> `Message`, `str` -> `bytes` (these IIRC are from an older similar issue).